airbyte-integrations/connectors/source-microsoft-onedrive/source_microsoft_onedrive/stream_reader.py

```diff
@@ -0,0 +1,171 @@
+#
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+#
+
+import logging
+from contextlib import contextmanager
+from functools import lru_cache
+from io import IOBase
+from typing import Iterable, List, Optional
+
+import smart_open
+from airbyte_cdk.sources.file_based.file_based_stream_reader import AbstractFileBasedStreamReader, FileReadMode
+from airbyte_cdk.sources.file_based.remote_file import RemoteFile
+from airbyte_cdk.utils.traced_exception import AirbyteTracedException, FailureType
+from msal import ConfidentialClientApplication
+from msal.exceptions import MsalServiceError
+from office365.graph_client import GraphClient
+from source_microsoft_onedrive.spec import SourceMicrosoftOneDriveSpec
+
+
+class MicrosoftOneDriveRemoteFile(RemoteFile):
+    download_url: str
+
+
+class MicrosoftOneDriveClient:
+    """
+    Client to interact with Microsoft OneDrive.
+    """
+
+    def __init__(self, config: SourceMicrosoftOneDriveSpec):
+        self.config = config
+        self._client = None
+
+    @property
+    @lru_cache(maxsize=None)
+    def msal_app(self):
+        """Returns an MSAL app instance for authentication."""
+        return ConfidentialClientApplication(
+            self.config.credentials.client_id,
+            authority=f"https://login.microsoftonline.com/{self.config.credentials.tenant_id}",
+            client_credential=self.config.credentials.client_secret,
+        )
+
+    @property
+    def client(self):
+        """Initializes and returns a GraphClient instance."""
+        if not self.config:
+            raise ValueError("Configuration is missing; cannot create the Office365 graph client.")
+        if not self._client:
+            self._client = GraphClient(self._get_access_token)
+        return self._client
+
+    def _get_access_token(self):
+        """Retrieves an access token for OneDrive access."""
+        scope = ["https://graph.microsoft.com/.default"]
+        refresh_token = self.config.credentials.refresh_token
+
+        if refresh_token:
```

Comment:
No, because it's oauth authentication and to proceed with config we need an refresh token, if you need authenticate without it they change authentication method